accepted/0000-dependency-selector-syntax.md

```diff
@@ -0,0 +1,421 @@
+
+#### Example of an `Object`: 
+```css
+/* return dependencies that have a `scripts.test` containing `"tap"` */
+*:scripts([test~=tap])
+```
+
+#### Example of an `Array` Attribute Selection:
+```css
+/* return dependencies that have a keyword that begins with "react" */
+*:keywords([value^="react"])
+```
+
+#### Example of an `Array` matching directly to a value:
+```css
+/* return dependencies that have the exact keyword "react" */
+/* this is equivalent to `*:keywords([value="react"])` */
+*:keywords("react")
+```
+
+#### Example of an `Array` of `Object`s: 
+```css
+/* returns */
+*:contributors([email="ruyadorno@github.com"])`
+```
+
+### Classes
+
+Given that `Arborist` will control our understanding of the DOM (Dependency Object Model), claseses are predefined by their relationships to their ancestors & are specific to their dependency type. Dependencies may have, & are allowed to have, multiple classifications (ex. a `workspace` may also be a `dev` dependency).
+
+- `.prod`
+- `.dev`
+- `.optional`
+- `.peer`
+- `.bundled`
+- `.workspace`
+
+### Attributes
+
+There are several known attributes that are normalized & queryable living in `Node.package` (aka. `package.json`).
+
+- `[name]`
+- `[version]`
+- `[description]`
+- `[homepage]`
+- `[bugs]`
+- `[author]`
+- `[license]`
+- `[funding]`
+- `[files]`
+- `[main]`
+- `[browser]`
+- `[bin]`
+- `[man]`
+- `[directories]`
+- `[repository]`
+- `[scripts]`
+- `[config]`
+- `[workspaces]`
+- `[dependencies]`
+- `[devDependencies]`
+- `[optionalDependencies]`
+- `[bundledDependencies]`
+- `[peerDependencies]`
+- `[peerDependenciesMeta]`
+- `[engines]`
+- `[os]`
+- `[cpu]`
+- `[keywords]`
+
+### Command
+
+#### `npm query "<selector>"` (alias `q`)
+
+#### Options:
+
+- `query-output`
+  - Default: `json`
+  - Type: `json`, `list`, `explain`, `outdated`, `funding`, `audit`, `duplicates`, `file`
+
+#### Response Output
+
+- an array of dependency objects is returned which can contain multiple copies of the same package which may or may not have been linked or deduped
+
+```json
+[
+  {
+    "name": "",
+    "version": "",
+    "description": "",
+    "homepage": "",
+    "bugs": {},
+    "author": {},
+    "license": {},
+    "funding": {},
+    "files": [],
+    "main": "",
+    "browser": "",
+    "bin": {},
+    "man": [],
+    "directories": {},
+    "repository": {},
+    "scripts": {},
+    "config": {},
+    "dependencies": {},
+    "devDependencies": {},
+    "optionalDependencies": {},
+    "bundledDependencies": {},
+    "peerDependencies": {},
+    "peerDependenciesMeta": {},
+    "engines": {},
+    "os": [],
+    "cpu": [],
+    "workspaces": {},
+    "keywords": [],
+    "ancenstry": "",
+    "path": "",
+    "realpath": "",
+    "parent": "",
+    "vulnerabilities": [],
+    "cwe": []
+  },
+  ...
```

Comment:
markdown parsing isn't happy with the `...` here. IMO fine to remove it :P

Reply:
Parser doesn't like it but it still renders it nicely... I'm okay with removing it though

Reply:
```suggestion
  …
```